### PR TITLE
Refactoring of the Bone class

### PR DESCRIPTION
**Pull request by Popov72**
It's mainly a reworking of the names, to make things a little clearer for the developer/end-user.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14007/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Review by sebavan — APPROVED** [review]

**Review by RaananW — APPROVED** [review]

**Review by bghgary — APPROVED** [review]
Some minor comments

File: packages/dev/core/src/Bones/bone.ts [hunk 1]
```diff
@@ -96,23 +97,28 @@ export class Bone extends Node {
...
         localMatrix: Nullable<Matrix> = null,
-        restPose: Nullable<Matrix> = null,
-        baseMatrix: Nullable<Matrix> = null,
+        restMatrix: Nullable<Matrix> = null,
+        bindMatrix: Nullable<Matrix> = null,
         index: Nullable<number> = null
     ) {
         super(name, skeleton.getScene());
         this._skeleton = skeleton;
         this._localMatrix = localMatrix ? localMatrix.clone() : Matrix.Identity();
```
[the remark this answers is not in the document]
Comment (Popov72) [reply packages/dev/core/src/Bones/bone.ts]: Done, but to keep backward compatibility, I'm still cloning the local matrix.

[the remark this answers is not in the document]
Comment (bghgary) [reply packages/dev/core/src/Bones/bone.ts]: Oops, yeah, I missed the cloning.

**Pull request merged**